tests/spec/core/dfn-spec.js

```diff
@@ -153,4 +153,75 @@ describe("Core — Definitions", () => {
       expect(el.hash).toBe("#dfn-first");
     }
   });
+
+  describe("internal slot definitions", () => {
+    const body = `
+      <section data-dfn-for="Test" data-cite="HTML">
+        <h2>Internal slots</h2>
+        <pre class="idl">
+          [Exposed=Window]
+          interface Foo{};
+        </pre>
+        <p>
+          <dfn id="attribute">
+            [[\\internal slot]]
+          </dfn>
+          <dfn id="method">
+            [[\\I am_a method]](I, really, ...am)
+          </dfn>
+          <dfn id="parent" data-dfn-for="Window">[[\\internal slot]]</dfn>
+          <dfn id="broken" data-dfn-for="">[[\\broken]]</dfn>
+        </p>
+        <section data-dfn-for="">
+          <h2>.</h2>
+          <p>
+            <dfn id="broken-parent">[[\\broken]]</dfn>
+          </p>
+        </section>
+        <p>
+        {{Test/[[internal slot]]}}
+        {{Test/[[I am_a method]](I, really, ...am)}}
+        {{Window/[[internal slot]]}}
+        </p>
+      </section>
+    `;
+
+    it("sets the data-dfn-type as an attribute", async () => {
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.getElementById("attribute");
+      expect(dfn.textContent.trim()).toBe("[[internal slot]]");
+      expect(dfn.dataset.dfnType).toBe("attribute");
+      expect(dfn.dataset.idl).toBe("");
+    });
+
+    it("sets the data-dfn-type as a method, when it's a method", async () => {
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.getElementById("method");
+      expect(dfn.textContent.trim()).toBe(
+        "[[I am_a method]](I, really, ...am)"
```

Comment:
Are spaces acceptable in internal slot names? Any real world examples?

Reply:
there are no rules, which is another reason why I'm not very comfortable with labeling them as attributes; there are real-world examples of spaces in internal slots, like [`[[unique id]]`](https://drafts.css-houdini.org/css-layout-api-1/#dom-layoutchild-unique-id-slot) (`term.startsWith('[[') && term.match(/ .+\]/)` lists them all on https://respec.org/xref/filter/)

Reply:
(I should've checked given I wrote the filter tool 😆)

Reply:
Though, at least no internalSlot-method with spaces.